### ISSUE TITLE
Typical workflow?

### ISSUE THREAD

**Issue by jessecusack** [gunnarvoet/gadcp#15]
I'm trying to understand the current workflow with this package to see where bin mapping might fit.

My understanding from the notebook is it goes something like this:

1. Initialise a `ProcessADCP` object. 
2. Check the editing parameters with convenient plotting functions
3. Do the editing
4. Save to netcdf via xarray (?)

The part that confuses me right now is step 3. It seems like this happens when `average_ensembles` or `burst_average_ensembles` is run, via calls to `_edit` etc. 

To fit into the current workflow, I could create a `_binmap` method and then call it in one of the averaging methods. Since I'm working with single ping data, not burst/ensemble (concepts I also don't fully understand), a lot of the averaging steps seem unnecessary. Is there some way of splitting the editing out into smaller chunks that could be mixed/matched as needed?

**Comment by gunnarvoet** [comment]
Yes, that describes the workflow pretty well. I wouldn't claim that this is a very intuitive or generally efficient way to step through the processing, it just matched what I was doing with the dataset. Happy to hear suggestions for how to change this. In general, it may make sense to differentiate a bit more between ping / ensemble / averaged & depth gridded dataset and not have everything inside one bloated class.

The burst sampling is really also just working with single pings but on a non-regular time grid, i.e. ping a few times and then wait for a while. I have not come to a conclusion whether generally this makes more or less sense than pinging at a regular interval but seem to gravitate towards it.

**Comment by gunnarvoet** [comment]
Yes, in general I agree to this. There are some subtleties, for example when working with pings from a burst sampling scheme it is advantageous to do some data editing and the percent good calculation *before* depth gridding while I think this is true the other way round for ensembles formed from pings that are further apart in time. I am sure there are ways to deal with this and agree that defining processing levels should be really helpful, both to write cleaner code and to make it easier for the user to understand what is happening.

**Comment by jessecusack** [comment]
Hmm. Maybe after splitting off a few of the processing steps into separate methods it will become clear how to handle the various options. I'll probably start by tackling bin mapping first.
